src/test/java/io/aiven/kafka/auth/AivenAclAuthorizerV2Test.java

```diff
@@ -110,78 +120,147 @@ void tearDown() {
+
+        checkSingleAction(requestCtx("User", "pass"), action(READ_OPERATION, TOPIC_RESOURCE), false);
 
         // check that config is reloaded after file modification
         Files.write(configFilePath, ACL_JSON_LONG.getBytes());
         Thread.sleep(100);
 
-        assertThat(auth.authorize(startSessionFor("User", "pass-1"), READ_OPERATION, TOPIC_RESOURCE)).isTrue();
+        checkSingleAction(requestCtx("User", "pass-1"), action(READ_OPERATION, TOPIC_RESOURCE), true);
 
         // check that config is reloaded after file deletion
         assertThat(configFilePath.toFile().delete()).isTrue();
         Thread.sleep(100);
 
-        assertThat(auth.authorize(startSessionFor("User", "pass-1"), READ_OPERATION, TOPIC_RESOURCE)).isFalse();
+        checkSingleAction(requestCtx("User", "pass-1"), action(READ_OPERATION, TOPIC_RESOURCE), false);
 
         // check that config is reloaded after directory deletion
         assertThat(Files.deleteIfExists(configFilePath.getParent().toAbsolutePath())).isTrue();
         Thread.sleep(100);
 
-        assertThat(auth.authorize(startSessionFor("User", "pass-1"), READ_OPERATION, TOPIC_RESOURCE)).isFalse();
+        checkSingleAction(requestCtx("User", "pass-1"), action(READ_OPERATION, TOPIC_RESOURCE), false);
 
         // check that config reloaded after file and directory re-creation
         assertThat(tmpDir.toFile().mkdir()).isTrue();
         Thread.sleep(100);
         Files.write(configFilePath, ACL_JSON.getBytes());
         Thread.sleep(100);
-        assertThat(auth.authorize(startSessionFor("User", "pass"), READ_OPERATION, TOPIC_RESOURCE)).isTrue();
+        checkSingleAction(requestCtx("User", "pass"), action(READ_OPERATION, TOPIC_RESOURCE), true);
+    }
+
+    private void startAndCheckAuthorizer() {
+        final AuthorizerServerInfo serverInfo = new AuthorizerServerInfo() {
+            @Override
+            public ClusterResource clusterResource() {
+                return new ClusterResource("cluster-id");
+            }
+
+            @Override
+            public int brokerId() {
+                return 0;
+            }
+
+            @Override
+            public Collection<Endpoint> endpoints() {
+                return List.of(
+                    new Endpoint("PLAINTEXT", SecurityProtocol.PLAINTEXT, "localhost", 9092),
+                    new Endpoint("SSL", SecurityProtocol.SSL, "localhost", 9093)
+                );
+            }
+
+            @Override
+            public Endpoint interBrokerEndpoint() {
+                return null;
+            }
+        };
+        assertThat(auth.start(serverInfo)).allSatisfy(
+            (endpoint, completionStage) -> {
+                // We just need this to not throw an exception.
+                try {
+                    completionStage.toCompletableFuture().get(0, TimeUnit.MICROSECONDS);
+                } catch (final Exception e) {
+                    throw new RuntimeException(e);
+                }
```

Comment:
Seems a bit easier and self-explanatory.

```suggestion
                assertThatNoException().isThrownBy(
                        () -> completionStage.toCompletableFuture().get(0, TimeUnit.MICROSECONDS));
```

Reply:
Good thing, done